storage-node/src/services/sync/workingProcess.ts

```diff
@@ -37,7 +37,7 @@ export class WorkingStack implements TaskSink, TaskSource {
   }
 
   async get(): Promise<SyncTask | null> {
-    const task = this.workingStack.pop()
+    const task = this.workingStack.pop() // expensive if array is very large?
```

Comment:
If implemented correctly pop is very fast as it only involves reducing array length and returning last element